lib/pacemaker/libpacemaker_private.h

```diff
@@ -231,6 +231,31 @@ struct resource_alloc_functions_s {
     void (*shutdown_lock)(pe_resource_t *rsc);
 };
 
+// HashTable
+
+typedef struct Node {
+    pe_resource_t* value;
+    struct Node *next;
+} Node;
+
+struct set
+{
+    int key;
+    Node* head;
+};
+
+G_GNUC_INTERNAL
+pe_resource_t * find(const char* key, struct set *array);
+
+G_GNUC_INTERNAL
+void insert(char* key, pe_resource_t* data, struct set *array);
+
+G_GNUC_INTERNAL
+void init_array(struct set **array);
+
+G_GNUC_INTERNAL
+void free_table(struct set *array);
+
```

Comment:
You can use the GHashTable type:
* init_array() -> GHashTable *table = pcmk__strkey_table(free, NULL)
* insert() -> g_hash_table_insert(table, key, value)
* find() -> g_hash_table_lookup(table, key)
* free_table() -> g_hash_table_destroy(table)